src/providers/info.rs

```diff
@@ -100,6 +101,9 @@ pub trait InfoProvider {
     async fn address_ex_from_any_source(&self, address: &str) -> ApiResult<AddressEx>;
     async fn address_ex_from_contracts(&self, address: &str) -> ApiResult<AddressEx>;
     fn chain_id(&self) -> &str;
+
+    fn client(&self) -> Arc<dyn HttpClient>;
+    fn cache(&self) -> Arc<dyn Cache>;
```

Comment:
Maybe it's better to add to the PR description the context of this change (since from the PR it seems mostly test related)

Reply:
Yeah, this is the part where the barrier gets a bit blurry, as this was needed so that we could use the same `info_provider` instance throughout the same request. 

I will document this in the description like you suggest 👍